From 6b74147fc8de0ada711175aa46876fc7d41b562f Mon Sep 17 00:00:00 2001
From: 
Date: Mon, 11 Jun 2018 04:08:38 -0500
Subject: [PATCH] add support for linux-musl to configure and remove glibc
 check

---
 configure.ac | 23 ++++-------------------
 1 file changed, 4 insertions(+), 19 deletions(-)

diff --git a/configure.ac b/configure.ac
index 848dc4ff3..666a9ee04 100644
--- a/configure.ac
+++ b/configure.ac
@@ -573,7 +573,7 @@ solaris*)
     fi
     ;;
 
-linux-gnu*|k*bsd*-gnu*)
+linux-gnu*|linux-musl*|k*bsd*-gnu*)
     test_gtk=yes
     build_gstreamer_1_0=yes
     build_gstreamer_0_10=yes
@@ -4012,7 +4012,7 @@ kfreebsd*)
     esac
     ;;
 
-linux-gnu*)
+linux-gnu*|linux-musl*)
     COM=GCC
     USING_X11=TRUE
     OS=LINUX
@@ -6957,7 +6957,7 @@ if test -n "$ENABLE_JAVA" -a -z "$JAVAINC"; then
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
         ;;
 
-    linux-gnu*)
+    linux-gnu*|linux-musl*)
         JAVAINC="-I$JAVA_HOME/include"
         JAVAINC="$JAVAINC -I$JAVA_HOME/include/linux"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
@@ -9460,21 +9460,6 @@ if test "$with_system_libexttextcat" = "yes"; then
 fi
 AC_SUBST(SYSTEM_LIBEXTTEXTCAT_DATA)
 
-dnl ***************************************
-dnl testing libc version for Linux...
-dnl ***************************************
-if test "$_os" = "Linux"; then
-    AC_MSG_CHECKING([whether libc is >= 2.1.1])
-    exec 6>/dev/null # no output
-    AC_CHECK_LIB(c, gnu_get_libc_version, HAVE_LIBC=yes; export HAVE_LIBC)
-    exec 6>&1 # output on again
-    if test "$HAVE_LIBC"; then
-        AC_MSG_RESULT([yes])
-    else
-        AC_MSG_ERROR([no, upgrade libc])
-    fi
-fi
-
 dnl =========================================
 dnl Check for uuidgen
 dnl =========================================
@@ -12470,7 +12455,7 @@ else
 
     case "$host_os" in
 
-    aix*|dragonfly*|freebsd*|linux-gnu*|*netbsd*|openbsd*)
+    aix*|dragonfly*|freebsd*|linux-gnu*|linux-musl*|*netbsd*|openbsd*)
         if test "$ENABLE_JAVA" != ""; then
             pathmunge "$JAVA_HOME/bin" "after"
         fi
-- 
2.17.1

