From 6e12927f734355f3f038719eba3fe068073b4b1b Mon Sep 17 00:00:00 2001
From: "Jory A. Pratt" <anarchy@gentoo.org>
Date: Tue, 12 Jun 2018 21:33:54 -0500
Subject: [PATCH] Add missing sys/types.h

Signed-off-by: Jory A. Pratt <anarchy@gentoo.org>
---
 lorcon_packasm.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lorcon_packasm.h b/lorcon_packasm.h
index 4dc91e7..a2fd4a3 100644
--- a/lorcon_packasm.h
+++ b/lorcon_packasm.h
@@ -155,6 +155,8 @@ void lcpa_free(struct lcpa_metapack *in_head);
 /* Get the size of an assembled LCPA packet */
 int lcpa_size(struct lcpa_metapack *in_head);
 
+#include <sys/types.h>
+
 /* Freeze a LCPA fragment packet into a bytestream.  The caller is responsible
  * for providing a bytestream of sufficient length. */
 void lcpa_freeze(struct lcpa_metapack *in_head, u_char *bytes);
-- 
2.17.1

