diff --git a/autogen.sh b/autogen.sh
index c894008..f5c6a7c 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,4 +1,3 @@
 #!/bin/sh
 # Run this to generate all the initial makefiles, etc.
-export ACLOCAL="aclocal -I missing_M4"
 autoreconf -f -i
diff --git a/configure.ac b/configure.ac
index f06488a..443aa8d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -152,6 +152,8 @@ dnl *********************************************************************
 dnl Test whether we're linux or not. If so, define HAVE_V4L and build the
 dnl video4linux/hardware dependent tools. Else, undefine HAVE_V4L
 
+AC_ARG_WITH([v4l], AC_HELP_STRING([--without-v4l], [Disable v4l1 API]))
+AS_IF([test "x$with_v4l" != "xno"], [
 have_video4linux=false
 case $host in
   *-*-linux*)
@@ -159,9 +161,9 @@ case $host in
                   [have_video4linux=true
                    AC_DEFINE(HAVE_V4L, 1,
 		             [Building for Linux - using the video4linux API])],
+                  []);; esac])
+AS_IF([test "x$have_video4linux" != "xtrue" && test "x$with_v4l" != "xno"],
                   [AC_MSG_ERROR([videodev.h not found - please install the linux kernel headers])])
-   ;;
-esac
 AM_CONDITIONAL(HAVE_V4L, test x$have_video4linux = xtrue)
 
 dnl *********************************************************************

